### PR TITLE
The annotation currently being edited is now clearly identifiable

### PR DESCRIPTION
**Pull request by kerryland**
When editing annotations it's now each to tell which piece of text is related to the comment.

A couple of notes:
- My change to `EditAnnotations::StartEditAnnotations` was simply because I was offended by the repetition of calls to `SetColor` :-)
- `EditAnnotations::SelectAnnotationInEditWindow` has a new call to `RepaintAsync`, which smells a bit to me. There might be a better way to achieve the same goal, but I don't know it.
- I apologise if this pull request includes some stupid merges from master, my git skills are a bit rusty

This change fixes another issue I raised in [issue 3638
](https://github.com/sumatrapdfreader/sumatrapdf/issues/3638)

image

### CODE REVIEW

**Comment by kerryland** [comment]
This fixes https://github.com/sumatrapdfreader/sumatrapdf/issues/1996

**Comment by kerryland** [comment]
I've realised, after reading https://github.com/sumatrapdfreader/sumatrapdf/issues/3590, that this change makes deleting an annotation annoying, so I will fix that first.

File: src/EditAnnotations.cpp [hunk 1]
```diff
@@ -1535,26 +1548,40 @@ Annotation* EngineMupdfCreateAnnotation(EngineBase* engine, AnnotationType typ,
 
     pdf_update_annot(ctx, annot);
     auto res = MakeAnnotationPdf(epdf, annot, pageNo);
+
+    std::optional <PdfColor> col;
```
Comment (kjk) [inline src/EditAnnotations.cpp:1552]: We don't use std:: if we can avoid it. It causes terrible bloat of executables.

Set to `ColorUnset` instead.

File: src/WindowTab.h [hunk 2]
```diff
@@ -64,6 +64,13 @@ struct WindowTab {
     bool IsDocLoaded() const;
     void MoveDocBy(int dx, int dy) const;
     void ToggleZoom() const;
+
+    struct {
+        bool show = false;
+        RectF rect;
+        int page = 0;
+        bool scrolled = false; // only automatically scroll once
+    } currentEditAnnotationMark;
```
Comment (kjk) [inline src/WindowTab.h:73]: To be consistent with our naming scheme, it should be `CurrentEditAnnotationMark`

**Pull request closed without merging**